client/src/components/datasetSelector/datasetSelector.tsx

```diff
@@ -0,0 +1,219 @@
+/* Core dependencies */
+import { Icon } from "@blueprintjs/core";
+import { IconNames } from "@blueprintjs/icons";
+import React, { FC, useEffect, useState } from "react";
+import { connect } from "react-redux";
+
+/* App dependencies */
+import { selectDataset } from "../../actions";
+import { DatasetMetadata, Dataset } from "../../common/types/entities";
+import DatasetMenu, { DatasetSelectedFn } from "./datasetMenu";
+import { AppDispatch, RootState } from "../../reducers";
+import { selectIsSeamlessEnabled } from "../../selectors/datasetMetadata";
+import TruncatingBreadcrumb from "./truncatingBreadcrumb";
+import TruncatingBreadcrumbs, {
+  TruncatingBreadcrumbProps,
+} from "./truncatingBreadcrumbs";
+
+/*
+ Actions dispatched by dataset selector.
+ */
+interface DispatchProps {
+  selectDataset: DatasetSelectedFn;
+}
+
+/*
+ Props selected from store.
+ */
+interface StateProps {
+  datasetMetadata: DatasetMetadata;
+  portalUrl: string;
+  seamlessEnabled: boolean;
+}
+
+type Props = StateProps & DispatchProps;
+
+/*
+ Map slice selected from store to props.
+ */
+const mapStateToProps = (state: RootState): StateProps => ({
+  datasetMetadata: state.datasetMetadata?.datasetMetadata,
+  portalUrl: state.datasetMetadata?.portalUrl,
+  seamlessEnabled: selectIsSeamlessEnabled(state),
+});
+
+/*
+ Map actions dispatched by dataset selector to props.
+ */
+const mapDispatchToProps = (dispatch: AppDispatch): DispatchProps => ({
+  selectDataset: (dataset: Dataset) => dispatch(selectDataset(dataset)),
+});
+
+// Inline styles for icons
+const STYLE_ICON = { marginLeft: "5px", marginRight: 0 };
+
+/*
+ App-level collection and dataset breadcrumbs.
+ */
+const DatasetSelector: FC<Props> = ({
+  datasetMetadata,
+  portalUrl,
+  seamlessEnabled,
+  selectDataset: selectDatasetFn,
+}) => {
+  // Don't render if seamless features are not available.
+  if (!seamlessEnabled) {
```

Comment:
React Hooks cannot be conditionally called, so we need to move this guard clause below the hooks!

This also made me realize that we need to add react hooks eslint plugin!

PR here 🙆‍♂️  Thank you!

https://github.com/chanzuckerberg/single-cell-explorer/pull/114

Reply:
Got it, thanks Timmy!